### DISCUSSION TITLE
Are comments allowed in ignore files?

### DISCUSSION THREAD

**Discussion by unphased** [BurntSushi/ripgrep#2457] [Q&A] [answered]
I can't know for sure because it isn't spelled out in the readme. 

I'd like to annotate my rg ignore file because there is no built in logic to make files tracked by git non-ignored if they match any globs that are specified in ignore files (or happen to be dotfiles).

So I need to take these exceptions and list them with a leading `!` in the ignore file. I'm collecting many tens of these by this point and would like to be able to jot down comments to explain them in the ignore file.

**Comment by BurntSushi** [comment, marked as answer]
The [GUIDE defers to `man gitignore`](https://github.com/BurntSushi/ripgrep/blob/master/GUIDE.md#automatic-filtering) for the precise syntax. And indeed, the gitignore format supports comments. From `man gitignore`:

> A line starting with `#` serves as a comment. Put a backslash ("\") in front of the first hash for patterns that begin with a hash.
